subsys/fs/shell.c

```diff
@@ -275,6 +275,76 @@ static int cmd_rm(const struct shell *sh, size_t argc, char **argv)
 	return err;
 }
 
+static int cmd_cp(const struct shell *sh, size_t argc, char **argv)
+{
+	int err;
+	char path_src[MAX_PATH_LEN];
+	char path_dst[MAX_PATH_LEN];
+	struct fs_file_t file_src;
+	struct fs_file_t file_dst;
+	uint8_t buf[BUF_CNT];
+	ssize_t buf_len;
+	uint8_t *write_buf;
+	ssize_t num_written;
+
+	create_abs_path(argv[1], path_src, sizeof(path_src));
+	create_abs_path(argv[2], path_dst, sizeof(path_dst));
+
+	fs_file_t_init(&file_src);
+	fs_file_t_init(&file_dst);
+
+	err = fs_open(&file_src, path_src, FS_O_READ);
+	if (err) {
+		shell_error(sh, "Failed to open %s (%d)", path_src, err);
+		goto exit;
+	}
+
+	err = fs_open(&file_dst, path_dst, FS_O_CREATE | FS_O_TRUNC | FS_O_WRITE);
+	if (err) {
+		shell_error(sh, "Failed to open %s (%d)", path_dst, err);
+		goto close_src;
+	}
+
+	while (true) {
+		buf_len = fs_read(&file_src, buf, BUF_CNT);
+		if (0 > buf_len) {
```

Comment:
I understand there is secure programming concept where you put literals/const  on the left to detect unwanted assignments, but this hurts to look, and is inconsistent with the rest of the file where variables are on the left of literal (argc comparisons).